src/client/ui/inputs/NumericInput.js

```diff
@@ -1,7 +1,162 @@
 import React from "react";
+import PropTypes from "prop-types";
+
 import styles from "./NumericInput.scss";
-import ReactNumericInput from "react-numeric-input";
 
-export default function NumericInput(props) {
-  return <ReactNumericInput style={false} className={styles.numericInput} {...props} />;
+function round(value) {
+  return Math.round(value * 1000) / 1000;
+}
+
+const partialValue = /[-.0]$/;
+const wholeNumber = /-?[0-9]+$/;
+
+export default class NumericInput extends React.Component {
+  static propTypes = {
+    value: PropTypes.number,
+    mediumStep: PropTypes.number,
+    smallStep: PropTypes.number,
+    bigStep: PropTypes.number,
+    onChange: PropTypes.func
+  };
+
+  static defaultProps = {
+    smallStep: 0.1,
+    mediumStep: 1,
+    bigStep: 10
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: props.value.toString(),
+      step: props.mediumStep
+    };
+    this.lastValidValue = this.props.value;
+  }
+
+  getStepForKeys(ctrlKey, shiftKey) {
+    let step = this.props.mediumStep;
+    if (ctrlKey) {
```

Comment:
ctrl + click on macOS is equivalent to a right click event. I think we may want to use the alt key instead. I think most programs do the smaller step with alt on mac. Is that also true for Windows?

Reply:
It looks like `ReactNumericInput` used `e.ctrlKey || e.metaKey` https://github.com/vlad-ignatov/react-numeric-input/blob/master/src/NumericInput.jsx#L707

I still kinda think alt makes more sense though. I use command to jump around inside inputs.

Reply:
Yeah, I tried `alt` but as far as I can tell, it always triggers the menu bar in Firefox and there's no way to prevent it. I'll try again, but if that doesn't work, maybe I can change it to metaKey for macOS alone.

Reply:
If you're going to go for something that check what platform you're on`alt`would be preferrable to `cmd` on mac. However, `e.ctrlKey || e.metaKey` would probably be just fine.